QtGraphicalEffects module is not available because it depends on OpenGL.
This patch removes all usage of QtGraphicalEffects (shadows and gradients).diff -uNr kirigami2-5.99.0/CMakeLists.txt kirigami2-5.99.0.mod/CMakeLists.txt
--- kirigami2-5.99.0/CMakeLists.txt	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/CMakeLists.txt	2022-11-12 15:59:23.578160679 +0800
@@ -128,7 +128,6 @@
 endif()
 
 
-ecm_find_qmlmodule(QtGraphicalEffects 1.0)
 
 ecm_set_disabled_deprecation_versions(
     QT 5.15.2
diff -uNr kirigami2-5.99.0/src/controls/Avatar.qml kirigami2-5.99.0.mod/src/controls/Avatar.qml
--- kirigami2-5.99.0/src/controls/Avatar.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/Avatar.qml	2022-11-12 15:59:54.918160667 +0800
@@ -7,7 +7,6 @@
 import QtQuick 2.13
 import org.kde.kirigami 2.14 as Kirigami
 import QtQuick.Controls 2.13 as QQC2
-import QtGraphicalEffects 1.0
 import org.kde.kirigami.private 2.14
 import "templates/private" as P
 
diff -uNr kirigami2-5.99.0/src/controls/ItemViewHeader.qml kirigami2-5.99.0.mod/src/controls/ItemViewHeader.qml
--- kirigami2-5.99.0/src/controls/ItemViewHeader.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/ItemViewHeader.qml	2022-11-12 16:00:50.808160646 +0800
@@ -6,7 +6,6 @@
 
 import QtQuick 2.5
 import QtQuick.Templates 2.0 as T2
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.4 as Kirigami
 import "private"
 
@@ -95,13 +94,6 @@
             elide: Text.ElideRight
 
             layer.enabled: root.backgroundImage.hasImage
-            layer.effect: DropShadow {
-                horizontalOffset: 0
-                verticalOffset: 2
-                radius: Kirigami.Units.smallSpacing*2
-                samples: 32
-                color: Qt.rgba(0, 0, 0, 0.7)
-            }
         }
     }
 }
diff -uNr kirigami2-5.99.0/src/controls/ScrollablePage.qml kirigami2-5.99.0.mod/src/controls/ScrollablePage.qml
--- kirigami2-5.99.0/src/controls/ScrollablePage.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/ScrollablePage.qml	2022-11-12 16:00:14.048160660 +0800
@@ -9,7 +9,6 @@
 import QtQuick.Templates 2.15 as T
 import QtQuick.Controls 2.15 as QQC2
 import QtQuick.Layouts 1.15
-import QtGraphicalEffects 1.0
 
 import org.kde.kirigami 2.19
 import org.kde.kirigami.templates 2.2 as KT
diff -uNr kirigami2-5.99.0/src/controls/private/ActionButton.qml kirigami2-5.99.0.mod/src/controls/private/ActionButton.qml
--- kirigami2-5.99.0/src/controls/private/ActionButton.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/private/ActionButton.qml	2022-11-12 16:03:44.848160579 +0800
@@ -7,7 +7,6 @@
 import QtQuick 2.15
 import QtQuick.Layouts 1.2
 import QtQuick.Controls 2.0 as Controls
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.16
 
 import "../templates/private"
@@ -433,15 +432,6 @@
                 }
             }
 
-            DropShadow {
-                anchors.fill: background
-                horizontalOffset: 0
-                verticalOffset: 1
-                radius: Units.gridUnit /2
-                samples: 16
-                color: Qt.rgba(0, 0, 0, mouseArea.pressed ? 0.6 : 0.4)
-                source: background
-            }
         }
     }
 
@@ -466,15 +456,6 @@
         height: width
 
 
-        DropShadow {
-            anchors.fill: handleGraphics
-            horizontalOffset: 0
-            verticalOffset: 1
-            radius: Units.gridUnit /2
-            samples: 16
-            color: Qt.rgba(0, 0, 0, fakeContextMenuButton.pressed ? 0.6 : 0.4)
-            source: handleGraphics
-        }
         Rectangle {
             id: handleGraphics
             anchors.fill: parent
diff -uNr kirigami2-5.99.0/src/controls/private/CornerShadow.qml kirigami2-5.99.0.mod/src/controls/private/CornerShadow.qml
--- kirigami2-5.99.0/src/controls/private/CornerShadow.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/private/CornerShadow.qml	2022-11-12 16:02:46.408160601 +0800
@@ -5,7 +5,6 @@
  */
 
 import QtQuick 2.1
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.4
 
 RadialGradient {
diff -uNr kirigami2-5.99.0/src/controls/private/EdgeShadow.qml kirigami2-5.99.0.mod/src/controls/private/EdgeShadow.qml
--- kirigami2-5.99.0/src/controls/private/EdgeShadow.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/private/EdgeShadow.qml	2022-11-12 16:06:58.898160505 +0800
@@ -5,10 +5,9 @@
  */
 
 import QtQuick 2.1
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.4
 
-LinearGradient {
+Item {
     id: shadow
     /**
      * @brief This property holds the edge of the shadow that will determine the direction of the gradient.
@@ -26,25 +25,5 @@
     implicitWidth: radius
     implicitHeight: radius
 
-    start: Qt.point((edge !== Qt.RightEdge ? 0 : width), (edge !== Qt.BottomEdge ? 0 : height))
-    end: Qt.point((edge !== Qt.LeftEdge ? 0 : width), (edge !== Qt.TopEdge ? 0 : height))
-    gradient: Gradient {
-        GradientStop {
-            position: 0.0
-            color: Qt.rgba(0, 0, 0, 0.25)
-        }
-        GradientStop {
-            position: 0.20
-            color: Qt.rgba(0, 0, 0, 0.1)
-        }
-        GradientStop {
-            position: 0.35
-            color: Qt.rgba(0, 0, 0, 0.02)
-        }
-        GradientStop {
-            position: 1.0
-            color:  "transparent"
-        }
-    }
 }
 
diff -uNr kirigami2-5.99.0/src/controls/templates/private/PassiveNotification.qml kirigami2-5.99.0.mod/src/controls/templates/private/PassiveNotification.qml
--- kirigami2-5.99.0/src/controls/templates/private/PassiveNotification.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/controls/templates/private/PassiveNotification.qml	2022-11-12 16:05:38.268160536 +0800
@@ -8,7 +8,6 @@
 import QtQuick.Controls 2.3 as Controls
 import QtQuick.Layouts 1.2
 import QtQuick.Window 2.2
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.12 as Kirigami
 
 /*
diff -uNr kirigami2-5.99.0/src/styles/Material/InlineMessage.qml kirigami2-5.99.0.mod/src/styles/Material/InlineMessage.qml
--- kirigami2-5.99.0/src/styles/Material/InlineMessage.qml	2022-10-01 22:20:15.000000000 +0800
+++ kirigami2-5.99.0.mod/src/styles/Material/InlineMessage.qml	2022-11-12 16:08:01.528160481 +0800
@@ -7,7 +7,6 @@
  */
 
 import QtQuick 2.7
-import QtGraphicalEffects 1.0
 import org.kde.kirigami 2.5 as Kirigami
 
 import "../../private"
